docs/src/components/ComponentDoc/ComponentExample/ComponentExample.js

```diff
@@ -64,36 +62,44 @@ class ComponentExample extends PureComponent {
     renderHtml: true,
   }
 
-  componentWillMount() {
-    const { examplePath } = this.props
-    this.anchorName = examplePathToHash(examplePath)
-
-    this.setState({
-      showCode: this.isActiveHash(),
-      sourceCode: this.getOriginalSourceCode(),
-    })
-  }
-
-  componentWillReceiveProps(nextProps) {
-    const { examplePath, exampleSources, location } = nextProps
-    const nextSourceCode = exampleSources[examplePath]
+  static getDerivedStateFromProps(nextProps, prevState = {}) {
```

Comment:
```suggestion
  static getDerivedStateFromProps(props, state) {
```

Let's use original naming proposed by React team

Reply:
Done